tests/middlewares/idempotent_tests.rb

```diff
@@ -161,6 +161,36 @@
     response.status
   end
 
+  tests("Idempotent request with custom error first 3 times").returns(200) do
```

Comment:
It would probably be good to also have a test that expects a failure, and raises an error which IS in the default error list, but is NOT in the specified list. I'm pretty confident the test will just pass already, but it may help guard us from regression. Does that make sense?

Reply:
Yes, that makes perfect sense since we want to make sure that retry_errors is not complementing the default list but rather completely overriding it. Spec added!